### PR TITLE
pyqbdi for python3 with pybind11

### PR DESCRIPTION
**Pull request by nsurbay**

### CODE REVIEW

File: tools/pyqbdi/CMakeLists.txt [hunk 1]
```diff
@@ -1,31 +1,105 @@
...
-    add_definitions("-DPYTHON_LIBRARIES=\"${PYTHON_LIBRARIES}\"")
 
-    add_library(pyqbdi SHARED ${SOURCES})
-    # Add build directory as include path for the mach_exc.h header
-    include_directories(pyqbdi PRIVATE
+    message(STATUS "Python lib:     ${PYTHON_LIBRARY}")
+    message(STATUS "Python include: ${PYTHON_INCLUDE_DIR}")
+    message(STATUS "Python interpreter: ${PYTHON_EXECUTABLE}")
+
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s -frtti -DQBDI_PLATFORM_STR='\"${QBDI_PLATFORM}\"' -DQBDI_ARCH_STR='\"${QBDI_ARCH}\"'")
```
Comment (romainthomas) [inline tools/pyqbdi/CMakeLists.txt]: Use ``target_compile_options()``

File: tools/pyqbdi/binding/Callback.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ */
+
+#include "pyqbdi.hpp"
+
+namespace QBDI {
+namespace pyQBDI {
+
+void init_binding_Callback(py::module& m) {
+
+    py::enum_<QBDI::VMAction>(m, "VMAction", "The callback results.")
```
Comment (romainthomas) [inline tools/pyqbdi/binding/Callback.cpp]: No need to use ``QBDI::`` as you are already in the namespace QBDI

File: tools/pyqbdi/binding/Callback.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                });
+
+    py::enum_<QBDI::MemoryAccessType>(m, "MemoryAccessType", "Memory access type (read / write / ...)", py::arithmetic())
+        .value("MEMORY_READ", QBDI::MemoryAccessType::MEMORY_READ, "Memory read access")
+        .value("MEMORY_WRITE", QBDI::MemoryAccessType::MEMORY_WRITE, "Memory write access")
+        .value("MEMORY_READ_WRITE", QBDI::MemoryAccessType::MEMORY_READ_WRITE, "Memory read/write access")
+        .export_values();
+
+    py::class_<QBDI::VMState>(m, "VMState")
+        .def(py::init<>())
```
Comment (romainthomas) [inline tools/pyqbdi/binding/Callback.cpp]: No need ctor

File: tools/pyqbdi/binding/Memory.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+namespace pyQBDI {
+
+void init_binding_Memory(py::module& m) {
+
+    py::enum_<QBDI::Permission>(m, "Permission", py::arithmetic(), "Memory access rights.")
+        .value("PF_NONE", QBDI::Permission::PF_NONE, "No access")
+        .value("PF_READ", QBDI::Permission::PF_READ, "Read access")
+        .value("PF_WRITE", QBDI::Permission::PF_WRITE, "Write access")
+        .value("PF_EXEC", QBDI::Permission::PF_EXEC, "Execution access")
+        .export_values()
```
Comment (romainthomas) [inline tools/pyqbdi/binding/Memory.cpp]: Move a the end

Comment (nsurbay) [reply tools/pyqbdi/binding/Memory.cpp]: `py::enum::def` return a `py::class`. The method `export_values` cannot be call on a `py::class`. We must call `export_values` before define `__str__` method.

**Comment by nsurbay** [comment]
Add wheel for:
- linux and windows (32 and 64 bits) for python 3.5, 3.6, 3.7 and 3.8
- macOS (64 bits) for python  3.6, 3.7 and 3.8

Add pyqbdipreload script (``python -m pyqbdipreload <script> <target> <args>..``).

uploaded (manually) on https://pypi.org/project/PyQBDI/

**Pull request merged**